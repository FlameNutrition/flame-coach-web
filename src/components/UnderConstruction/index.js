import {
  Box, Card, makeStyles, Typography
} from '@material-ui/core';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import React from 'react';
import UnderConstructionIcon from '../../images/UnderConstruction';
import { useIsMediumMobile, useIsMobile } from '../../utils/mediaUtil';

const useStyles = makeStyles(() => ({
  root: {
    height: '80vh'
  },
  rootMobileSmall: {
    height: '100vh'
  },
  rootMobile: {
    height: '80vh'
  },
  imageLayout: {
    marginTop: '100px',
    width: '100%'
  },
  messageLayout: {
    marginTop: '30px',
    width: '100%'
  },
  message: {
    paddingLeft: '20px',
    paddingRight: '20px',
  },
  submessage: {
    marginTop: '10px',
    paddingLeft: '20%',
    paddingRight: '20%',
  },
  imageMobile: {
    height: 'auto',
    width: '50%'
  },
  image: {
    height: '200pt',
    width: '200pt'
  }
}));

const WorkInProgress = ({ message, submessage }) => {
  const classes = useStyles();
  const isMobile = useIsMobile();
  const isMediumMobile = useIsMediumMobile();

  let rootClass = classes.root;

  if (isMobile) {
    if (isMediumMobile) {
      rootClass = classes.rootMobileSmall;
    } else {
      rootClass = classes.rootMobile;
    }
  }

  return (
    <>
      <Card
        className={rootClass}
      >
        <Box
          display="flex"
          justifyContent="center"
          flexWrap="wrap"
          flexDirection="row"
        >
          <Box
            display="flex"
            justifyContent="center"
            className={classes.imageLayout}
          >
            <UnderConstructionIcon className={isMobile ? classes.imageMobile : classes.image} viewBox="0 0 496 496" />
          </Box>
          <Box
            display="flex"
            flexDirection="column"
            justifyContent="center"
            className={classes.messageLayout}
          >
            <Typography
              variant="h3"
              align="center"
              className={classes.message}
            >
              {message}
            </Typography>
            <Typography
              variant="body1"
              align="center"
              className={clsx(classes.message, classes.submessage)}
            >
              {submessage}
            </Typography>
          </Box>
        </Box>
      </Card>
    </>
  );
};

WorkInProgress.propTypes = {
  message: PropTypes.string.isRequired,
  submessage: PropTypes.string
};

WorkInProgress.defaultProps = {};

export default WorkInProgress;
