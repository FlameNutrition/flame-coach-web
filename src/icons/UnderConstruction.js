import React from 'react';
import { SvgIcon } from '@material-ui/core';

const UnderConstruction = (props) => {
  return (
    <SvgIcon {...props}>
      <g>
        <path
          fill="#D4D2D3"
          d="M66.823,41.807h-28.19c-8.133,0-14.725,6.593-14.725,14.725v401.711c0,6.6,5.351,11.95,11.95,11.95 h33.74c6.6,0,11.95-5.351,11.95-11.95V56.532C81.548,48.399,74.955,41.807,66.823,41.807z"
        />
        <path
          fill="#D4D2D3"
          d="M445.177,41.807h28.19c8.133,0,14.725,6.593,14.725,14.725v401.711c0,6.6-5.351,11.95-11.95,11.95 h-33.74c-6.6,0-11.95-5.351-11.95-11.95V56.532C430.452,48.399,437.045,41.807,445.177,41.807z"
        />
      </g>
      <g>
        <polygon fill="#465059" points="74.425,78.967 25.806,101.334 0,153.394 0,233.703 101.506,171.355 154.735,78.967 " />
        <polygon
          fill="#465059"
          points="228.749,78.967 101.506,171.355 43.974,263.742 124.284,263.742 240.516,188.731 309.058,78.967 "
        />
        <polygon
          fill="#465059"
          points="383.071,78.967 240.516,188.731 198.296,263.742 278.606,263.742 395.303,184.671 463.381,78.967 "
        />
        <polygon
          fill="#465059"
          points="383.071,201.118 352.62,263.742 432.929,263.742 490.323,233.703 512,184.671 512,104.361"
        />
      </g>
      <g>
        <path fill="#FFCD38" d="M512,184.671v55.164c0,13.203-10.705,23.907-23.907,23.907h-55.164L512,184.671z" />
        <path
          fill="#FFCD38"
          d="M512,102.875v1.487l-159.38,159.38h-74.013L463.381,78.967h24.712 C501.295,78.967,512,89.672,512,102.875z"
        />
        <polygon fill="#FFCD38" points="383.071,78.967 198.296,263.742 124.284,263.742 309.058,78.967 " />
        <path fill="#FFCD38" d="M228.749,78.967L43.974,263.742H23.907C10.705,263.742,0,253.037,0,239.834v-6.132L154.735,78.967H228.749z" />
        <path fill="#FFCD38" d="M74.425,78.967L0,153.394v-50.519c0-13.203,10.705-23.907,23.907-23.907H74.425z" />
      </g>
      <g>
        <rect x="23.911" y="263.737" fill="#BAB8B9" width="57.642" height="31.101" />
        <rect x="430.451" y="263.737" fill="#BAB8B9" width="57.642" height="31.101" />
      </g>
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
      <g />
    </SvgIcon>
  );
};

export default UnderConstruction;
