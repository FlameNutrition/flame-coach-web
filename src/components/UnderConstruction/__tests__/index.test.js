import React from 'react';
import { render } from '../../../testing/test-utils';
import UnderConstruction from '../index';

describe('<UnderConstruction / >', () => {
  it('create component', () => {
    const { container } = render(
      <UnderConstruction
        message="Ops...this page is under construction!"
        submessage="Sorry, we are doing the best for having good features available for you. But unfortunately this is a free application and we don't have much time as we liked to work on it."
      />
    );

    expect(container).toMatchSnapshot();
    expect(container.querySelector('h3')).toHaveTextContent('Ops...this page is under construction!');
    expect(container.querySelector('p')).toHaveTextContent('Sorry, we are doing the best for having good features available for you. But unfortunately this is a free application and we don\'t have much time as we liked to work on it.');
    expect(container.querySelector('svg')).not.toBeNull();
  });
});
