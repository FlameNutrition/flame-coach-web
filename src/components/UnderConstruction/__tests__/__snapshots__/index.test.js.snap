// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnderConstruction / > create component 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-10"
    >
      <div
        class="MuiBox-root MuiBox-root-11 makeStyles-imageLayout-4"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-image-9"
          focusable="false"
          viewBox="0 0 496 496"
        >
          <g>
            <g>
              <path
                d="m496.769 104.908-15.321-62.548h-25.667l-15.321 62.548z"
                fill="#f9ef63"
              />
            </g>
            <g>
              <path
                d="m71.54 104.908-15.321-62.548h-25.667l-15.321 62.548z"
                fill="#f9ef63"
              />
            </g>
            <g>
              <path
                d="m481.448 42.36h-16.835l15.321 62.548h16.835z"
                fill="#f4dd45"
              />
            </g>
            <g>
              <path
                d="m56.219 42.36h-16.835l15.321 62.548h16.835z"
                fill="#f4dd45"
              />
            </g>
            <g>
              <circle
                cx="468.615"
                cy="30.931"
                fill="#e7a561"
                r="24.044"
              />
            </g>
            <g>
              <circle
                cx="43.386"
                cy="30.931"
                fill="#e7a561"
                r="24.044"
              />
            </g>
            <g>
              <path
                d="m474.372 7.586c.453 1.845.699 3.772.699 5.757 0 13.279-10.765 24.044-24.044 24.044-1.985 0-3.912-.245-5.757-.699 2.581 10.498 12.051 18.287 23.345 18.287 13.279 0 24.044-10.765 24.044-24.044 0-11.294-7.789-20.764-18.287-23.345z"
                fill="#e4936b"
              />
            </g>
            <g>
              <path
                d="m49.143 7.586c.453 1.845.699 3.772.699 5.757 0 13.279-10.765 24.044-24.044 24.044-1.985 0-3.912-.245-5.757-.699 2.58 10.498 12.051 18.287 23.345 18.287 13.279 0 24.044-10.765 24.044-24.044 0-11.294-7.789-20.764-18.287-23.345z"
                fill="#e4936b"
              />
            </g>
            <g>
              <path
                d="m236.581 135.885h36.657v427.408h-36.657z"
                fill="#f6f9f9"
                transform="matrix(0 1 -1 0 604.5 94.679)"
              />
            </g>
            <g>
              <path
                d="m433.706 331.261v3.841c0 7.541-6.113 13.654-13.654 13.654h-378.846v19.163h427.408v-36.657h-34.908z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m56.714 413.516h-26.657c-2.761 0-5-2.239-5-5v-297.883h36.657v297.883c0 2.762-2.238 5-5 5z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m481.943 413.516h-26.657c-2.761 0-5-2.239-5-5v-297.883h36.657v297.883c0 2.762-2.238 5-5 5z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m44.423 110.633v280.503c0 2.81-2.278 5.089-5.089 5.089h-14.277v12.291c0 2.761 2.239 5 5 5h26.657c2.761 0 5-2.239 5-5v-297.883z"
                fill="#b9c1cb"
              />
            </g>
            <g>
              <path
                d="m469.652 110.633v280.503c0 2.81-2.278 5.089-5.089 5.089h-14.278v12.291c0 2.761 2.239 5 5 5h26.657c2.761 0 5-2.239 5-5v-297.883z"
                fill="#b9c1cb"
              />
            </g>
            <g>
              <path
                d="m504 167.181h-34.297l-9.422-4.9-9.996 4.9h-442.285c-4.418 0-8-3.582-8-8v-63.765c0-4.418 3.582-8 8-8h496c4.418 0 8 3.582 8 8v63.765c0 4.418-3.582 8-8 8z"
                fill="#f6f9f9"
              />
            </g>
            <g>
              <path
                d="m504 87.416h-15.643v47.98c0 4.497-3.645 8.142-8.142 8.142h-480.215v15.643c0 4.418 3.582 8 8 8h496c4.418 0 8-3.582 8-8v-63.765c0-4.418-3.582-8-8-8z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m133.961 87.416h-64.768l-69.193 69.193v2.572c0 4.418 3.582 8 8 8h46.196z"
                fill="#ecbe6b"
              />
            </g>
            <g>
              <path
                d="m183.733 167.181 79.765-79.765h-64.768l-79.765 79.765z"
                fill="#ecbe6b"
              />
            </g>
            <g>
              <path
                d="m313.27 167.181 79.765-79.765h-64.768l-79.765 79.765z"
                fill="#ecbe6b"
              />
            </g>
            <g>
              <path
                d="m378.039 167.181h64.768l69.193-69.193v-2.572c0-4.418-3.582-8-8-8h-46.196z"
                fill="#ecbe6b"
              />
            </g>
            <g>
              <g>
                <path
                  d="m401.682 143.538-23.643 23.643h64.768l23.643-23.643z"
                  fill="#e7a561"
                />
              </g>
              <g>
                <path
                  d="m504 87.416h-15.643v34.215l23.643-23.643v-2.572c0-4.418-3.582-8-8-8z"
                  fill="#e7a561"
                />
              </g>
            </g>
            <g>
              <path
                d="m272.145 143.538-23.643 23.643h64.768l23.643-23.643z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m142.608 143.538-23.643 23.643h64.768l23.643-23.643z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m13.071 143.538-13.071 13.071v2.572c0 4.418 3.582 8 8 8h46.196l23.643-23.643z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m504 285.663h-496c-4.418 0-8-3.582-8-8v-63.765c0-4.418 3.582-8 8-8h496c4.418 0 8 3.582 8 8v63.765c0 4.418-3.582 8-8 8z"
                fill="#ecbe6b"
              />
            </g>
            <g>
              <path
                d="m504 205.898h-15.643v47.98c0 4.497-3.645 8.142-8.142 8.142h-480.215v15.643c0 4.418 3.582 8 8 8h496c4.418 0 8-3.582 8-8v-63.765c0-4.419-3.582-8-8-8z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m133.961 205.898h-64.768l-69.193 69.193v2.572c0 4.418 3.582 8 8 8h46.196z"
                fill="#f6f9f9"
              />
            </g>
            <g>
              <path
                d="m183.733 285.663 79.765-79.765h-64.768l-79.765 79.765z"
                fill="#f6f9f9"
              />
            </g>
            <g>
              <path
                d="m313.27 285.663 79.765-79.765h-64.768l-79.765 79.765z"
                fill="#f6f9f9"
              />
            </g>
            <g>
              <path
                d="m378.039 285.663h64.768l69.193-69.193v-2.572c0-4.418-3.582-8-8-8h-46.196z"
                fill="#f6f9f9"
              />
            </g>
            <g>
              <g>
                <path
                  d="m401.682 262.02-23.643 23.643h64.768l23.643-23.643z"
                  fill="#dae4ea"
                />
              </g>
              <g>
                <path
                  d="m504 205.898h-15.643v34.215l23.643-23.643v-2.572c0-4.419-3.582-8-8-8z"
                  fill="#dae4ea"
                />
              </g>
            </g>
            <g>
              <path
                d="m272.145 262.02-23.643 23.643h64.768l23.643-23.643z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m142.608 262.02-23.643 23.643h64.768l23.643-23.643z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m13.071 262.02-13.071 13.071v2.572c0 4.418 3.582 8 8 8h46.196l23.643-23.643z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m335.895 288-35.252 9.134-47.651 194.536h165.807l-47.651-194.536z"
                fill="#dae4ea"
              />
            </g>
            <g>
              <path
                d="m334.208 290.245 49.338 201.425h35.253l-47.651-194.536z"
                fill="#b9c1cb"
              />
            </g>
            <g>
              <path
                d="m434.117 505.113h-196.443c-2.209 0-4-1.791-4-4v-25.161c0-2.209 1.791-4 4-4h196.442c2.209 0 4 1.791 4 4v25.161c.001 2.209-1.79 4-3.999 4z"
                fill="#e7a561"
              />
            </g>
            <path
              d="m434.117 471.951h-13.104v12.058c0 2.209-1.791 4-4 4h-183.339v13.103c0 2.209 1.791 4 4 4h196.442c2.209 0 4-1.791 4-4v-25.161c.001-2.209-1.79-4-3.999-4z"
              fill="#e4936b"
            />
            <g>
              <path
                d="m322.127 242.027c-4.688 0-8.77 3.202-9.885 7.756l-11.598 47.35h70.505l-11.598-47.35c-1.115-4.554-5.197-7.756-9.885-7.756z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m271.694 415.318h128.402l-14.474-59.092h-99.453z"
                fill="#e7a561"
              />
            </g>
            <g>
              <path
                d="m324.297 249.783 11.598 47.35h35.252l-11.598-47.35c-1.115-4.554-5.197-7.756-9.885-7.756h-27.538c-1.357 0-2.659.276-3.857.767 2.942 1.208 5.235 3.754 6.028 6.989z"
                fill="#e4936b"
              />
            </g>
            <g>
              <path
                d="m364.844 415.318h35.252l-14.474-59.092h-35.252z"
                fill="#e4936b"
              />
            </g>
          </g>
        </svg>
      </div>
      <div
        class="MuiBox-root MuiBox-root-12 makeStyles-messageLayout-5"
      >
        <h3
          class="MuiTypography-root makeStyles-message-6 MuiTypography-h3 MuiTypography-alignCenter"
        >
          Ops...this page is under construction!
        </h3>
        <p
          class="MuiTypography-root makeStyles-message-6 makeStyles-submessage-7 MuiTypography-body1 MuiTypography-alignCenter"
        >
          Sorry, we are doing the best for having good features available for you. But unfortunately this is a free application and we don't have much time as we liked to work on it.
        </p>
      </div>
    </div>
  </div>
</div>
`;
